### ISSUE TITLE
[Suggestion] API Additions

### ISSUE THREAD

**Issue by luwie556** [Hazebyte/CrateReloaded#377]
```
CrateGiveEvent - Called when someone gives a crate to a player 
e.getSender(); -> Who sent the key.
e.getTarget();  -> Who got the key
e.getCrate() - Return the crate object of the key.
e.getAmount(). 
e.isKeyAll() - Check if its a keyall.

```
```
CrateReceiveEvent - Called when you are given a key.
e.getFrom(); - who its from.
e.getCrate() - Crate object. 
e.getAmount(); amount.
e.fromKeyAll(); 

```
Allow both events to be cancellable.

Also add some events that are commonly done in the plugin so that it would be a lot easier for developers to integrate into the plugin without having to go and make really roundabout ways to get to what they wanted when it could be made really easy.

**Comment by imWillX** [comment]
@luwie556 Is there a reason for the receive event?

The give event seems to accomplish the same task.

**Comment by luwie556** [comment]
If the give event is called practically every time a crate is given through command/API of the plugin. Than the answer would be no, you wouldn't have to go and create the receive event.

**Comment by imWillX** [comment]
Cool, would you need any other events?

**Comment by luwie556** [comment]
uh, I don't think so,

potentially an ability to open a crate manually like through API, almost like a "CrateAPI.openCrate(String crateName) or something

**Comment by imWillX** [comment]
Sure, we already have those.

https://github.com/Hazebyte/CrateReloadedAPI/blob/dev/src/main/java/com/hazebyte/crate/api/crate/CrateRegistrar.java#L90

https://github.com/Hazebyte/CrateReloadedAPI/blob/dev/src/main/java/com/hazebyte/crate/api/crate/Crate.java#L173

**Comment by luwie556** [comment]
I thought so, I wasn't sure. Thanks for adding those events.
Just a question will that API work for both premium and free resource or just Premium?

**Comment by imWillX** [comment]
The API should work for both. These calls don't work right now but are available via the API. I need to push the implementation in order for those API calls to work.

That being said, the implementation for the free version is one version behind the PRO. So the free version won't be receiving what I am adding in these next few days until the next release.

**Comment by luwie556** [comment]
Alright, I have the pro, so it doesn't matter either way. I just am wanting to go and make what i was planning to make a month ago when this API addition gets properly added. 

Thanks for adding this suggestion.

**Comment by imWillX** [comment]
Will be released in `2.0.9`
